### DISCUSSION TITLE
Google Cloud Build instructions & potential workaround for Google App Engine

### DISCUSSION THREAD

**Discussion by benatkin** [pnpm/pnpm#6034] [General]
I needed to run pnpm on Google Cloud Build and I was looking at this question from another person:

https://stackoverflow.com/questions/63206481/how-to-use-pnpm-with-google-cloud-build

I was able to figure out my issue right away, because I was just running `pnpm install` and `pnpm build` directly, just by running them in a single command:

There are [docs for other CIs](https://pnpm.io/continuous-integration) but Google Cloud Build is missing. I'd like to add it but am not sure my solution of running them in a single step is acceptable, and when I tried running `npm install -g pnpm` beforehand, that didn't work. I know I could use another container but I try to use official container images to reduce the amount of vetting of container images I need to do. So I may contribute to the docs when I have time, but in the meantime leave it here. Here is my build step:

```yaml
- name: node
  entrypoint: bash
  args:
  - -c
  - npm install -g pnpm && pnpm install && pnpm build
  dir: 'admin/frontend'
```

 I didn't have a good answer for `gcloud`, until I realized that running `pnpm` in a previous step and generating `package-lock.json` with `npm shrinkwrap` may work. After that, I assume gcloud would use. If I give it a shot myself I'll try to update this thread.

Here's a discussion about exporting `package-lock.json` which I agree is something that shouldn't be encouraged but I think in a CI that using `npm shrinkwrap` might be a good idea.

https://github.com/pnpm/pnpm/discussions/3367

**Comment by ThibaultJanBeyer** [comment]
Did you have any success?
